### DISCUSSION TITLE
Linux / Firefox: Video DownloadHelper fails to recover after suspend to ram

### DISCUSSION THREAD

**Discussion by ajfre** [aclap-dev/video-downloadhelper#978] [Help and Q&A]
### Your question

How to reproduce the bug:
- Start a download
- Set the system in suspend to RAM
- Restart
- The download hangs for ever in the state before entering supend to RAM

### Addon & CoApp version and details

Version 9.0.2.12

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

I also know diddly squat about Linux.  Probably less.  But is the "suspend to RAM" feature something like putting the system into sleep mode or hibernation mode?  In that case, I am going to guess that while your system is suspended, the download will time out after some delay.  The server will close your connection.  VDH has never been able to resume interrupted downloads from the point of failure.  Sadly, the only thing you can do is make sure your inflight downloads are complete before you suspend your system.  Either that or cancel whatever is running, make note of what that or those is/are, & suspend your system.  Later, when you resurrect your system, reload the relevant web pages & launch those downloads again.  Those will all be from the start.  That's the way VDH works.  Either you get the whole thing, or you have to retry from the start.  There is no in-between option.
